File: app/src/main/java/com/katdmy/android/unocount/PlayerViewModel.java
package com.katdmy.android.unocount;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.ArrayList;
import java.util.List;

public class PlayerViewModel extends AndroidViewModel {

    private PlayerRepository mPlayerRepository;

    private LiveData<List<Player>> mPlayers;

    private LiveData<List<String>> mActivePlayers;

    public PlayerViewModel(Application application) {
        super(application);
        mPlayerRepository = new PlayerRepository(application);
        mPlayers = mPlayerRepository.getPlayers();
        mActivePlayers = mPlayerRepository.getActivePlayers();
    }

    LiveData<List<Player>> getPlayers() {
        return mPlayers;
    }

    LiveData<List<String>> getActivePlayers() {
        return mActivePlayers;
    }

    void setActive(String playerName, boolean active) {
        mPlayerRepository.setActive(playerName, active);
    }

    void insertPlayer(Player player) {
        mPlayerRepository.addPlayer(player);
    }

    void deletePlayer(Player player) {
        mPlayerRepository.deletePlayer(player);
    }
}

File: app/src/main/java/com/katdmy/android/unocount/ui/player/PlayerViewModel.kt
package com.katdmy.android.unocount.ui.player

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.katdmy.android.unocount.domain.player.PlayerRepository
import com.katdmy.android.unocount.domain.player.model.Player

class PlayerViewModel(
        private val playerRepository: PlayerRepository
) : ViewModel() {
    var players: LiveData<List<Player>> = playerRepository.players
    var activePlayers: LiveData<List<String>> = playerRepository.activePlayers

    fun setActive(playerName: String, active: Boolean) {
        playerRepository.setActive(playerName, active)
    }

    fun insertPlayer(playerName: String) {
        playerRepository.addPlayer(playerName)
    }

    fun deletePlayer(playerName: String) {
        playerRepository.deletePlayer(playerName)
    }
}